### ISSUE TITLE
Feature request: option to output manifold in ImageJ plugin

### ISSUE THREAD

**Issue by tlambert03** [biocompibens/SME#6]
Would be nice to have an option to output the manifold z map (i.e. the Manifold array provided in the output of the Matlab functions) in the ImageJ plugin.  Perhaps just a checkbox that, if checked, outputs the manifold as an additional image?

**Comment by erexhepa** [comment]
Hi,

Yeah no problems with that. It was a feature of earlier version for debugging purposes. We also had as feature the save locally as a tiff the zmap. I can activate it and put it through my github account to leave the official version as it was described in the publication.
you have to be careful when manipulating it as depending on number of Z planes you need to adjust the format of the image so that it can be visualized optimally. Although till 256 z-levels it should be ok.

**Comment by tlambert03** [comment]
yeah, that'd be great thanks!  so, just keep an eye on https://github.com/erexhepa/SME?

**Comment by erexhepa** [comment]
yes keep an eye my local rep. I will post a comment here too so that you get notified.

**Comment by SoleneW** [comment]
Hello, 
I am also interested by this feature. Did you have time to implement it? I 

Thanks, 
Solène

**Comment by erexhepa** [comment]
No and I have to apologize because it is actually easily done and I can see it helping you guys out. I will do it today or tomorrow and let you guys know. Apologies again.

**Comment by SoleneW** [comment]
Thank you very much. It will indeed help me saving a time for further analysis.

**Comment by erexhepa** [comment]
Hi,

I uploaded the changes with regard to the interface.

I modified the interface and introduced two new text editable fields (at the bottom). One for the file name and the second one for the folder. Both fields are initialised with 1) the folder where the input image is located and the output image constructed as follow <input image filename><_smeprojection.tiff>. However you can this fields by adding text before the "_". The "smeprojection.tiff" extention however should be left unchanged because it is replace with "_smeprojectionImage.tiff" and "_smeprojectionIndex.tiff". The first image is the 2D projection and the second one is the index (depth) of the manifold. 

The plugin you can find it here :

https://github.com/erexhepa/IJ_PROJECTION_PLUGIN/tree/master/IJ2 
(SME_stacking.jar)

I have put also two examples images in the folder above to illustrate the outputs:
Centriole1_smeprojectionImage.tiff
Centriole1_smeprojectionIndex.tiff

I did some test but please let me know if there is something not working.

**Comment by SoleneW** [comment]
Thank you,

I will try it today an give you some feedbacks.



2018-04-21 19:36 GMT+02:00 Elton Rexhepaj <notifications@github.com>:

**Comment by SoleneW** [comment]
Hi, 
I tried to use the plugin with the updated functionality several time but I always end up with the same error which is related to imput/output permission denied. (see below). I tried changing the permission of the folder in which the image is and also the permission of the image but it didn't worked. 
Do you have any idea?

Thank you, 
Solène
![screenshot from 2018-04-23 13-20-52](https://user-images.githubusercontent.com/12595694/39177775-795ab1c0-47b0-11e8-8321-f77df4110f64.png)

**Comment by erexhepa** [comment]
Hi,

Sorry, I had forgot to retrieve the paths from the GUI so it was writing to the same folder where the input image is, hence the error as it seems the folder you have only read access. 

I have updated the plugin and it should be OK now. Please download it and try again. Just be careful with the / (Linux & Mac) and \ (Windows) (just check the path displayed by default on the GUI to be sure which one you should use).

Elton

**Comment by SoleneW** [comment]
Thanks,
it's working.

2018-04-25 7:33 GMT+02:00 Elton Rexhepaj <notifications@github.com>:
